core/src/main/java/io/questdb/griffin/engine/table/ShowCreateMatViewRecordCursorFactory.java

```diff
@@ -0,0 +1,216 @@
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8StringSink;
+import org.jetbrains.annotations.NotNull;
+
+import static io.questdb.griffin.engine.table.ShowCreateTableRecordCursorFactory.inVolumeToSink;
+import static io.questdb.griffin.engine.table.ShowCreateTableRecordCursorFactory.ttlToSink;
+
+public class ShowCreateMatViewRecordCursorFactory extends AbstractRecordCursorFactory {
+    public static final int N_DDL_COL = 0;
+    private static final RecordMetadata METADATA;
+    protected final TableToken tableToken;
+    protected final int tokenPosition;
+    private final ShowCreateMatViewCursor cursor = new ShowCreateMatViewCursor();
+
+    public ShowCreateMatViewRecordCursorFactory(TableToken tableToken, int tokenPosition) {
+        super(METADATA);
+        this.tableToken = tableToken;
+        this.tokenPosition = tokenPosition;
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
+        return cursor.of(executionContext, tableToken, tokenPosition);
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("show_create_materialized_view");
+        sink.meta("of").val(tableToken.getTableName());
+    }
+
+    public static class ShowCreateMatViewCursor implements NoRandomAccessRecordCursor {
+        protected final Utf8StringSink sink = new Utf8StringSink();
+        private final ShowCreateMatViewRecord record = new ShowCreateMatViewRecord();
+        protected SqlExecutionContext executionContext;
+        protected MatViewDefinition matViewDefinition;
+        protected CairoTable table;
+        private boolean hasRun;
+        private TableToken tableToken;
+
+        @Override
+        public void close() {
+            sink.clear();
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (!hasRun) {
+                sink.clear();
+                final CairoConfiguration config = executionContext.getCairoEngine().getConfiguration();
+                showCreateMatView(config);
+                hasRun = true;
+                return true;
+            }
+            return false;
+        }
+
+        public ShowCreateMatViewCursor of(
+                SqlExecutionContext executionContext,
+                TableToken tableToken,
+                int tokenPosition
+        ) throws SqlException {
+            this.tableToken = tableToken;
+            this.executionContext = executionContext;
+            try (MetadataCacheReader metadataRO = executionContext.getCairoEngine().getMetadataCache().readLock()) {
```

Comment:
nit: this is a confused code, the factory is passed a token, which is validated (exists) and matView. Not sure what the code below is attempting to handle? Table dropped before SQL execution? If so, engine.isDropped() does the trick.

Reply:
> Not sure what the code below is attempting to handle?

Which LoC do you mean exactly? The code seems to be reading table metadata from the metadata cache and then validates what was read from it.